plugins/base/src/main/kotlin/translators/documentables/DefaultPageCreator.kt

```diff
@@ -611,6 +630,19 @@ internal val Documentable.customTags: Map<String, SourceSetDependent<CustomTagWr
 private val Documentable.hasSeparatePage: Boolean
     get() = this !is DTypeAlias
 
+/**
+ * @see DefaultPageCreator.sortDivergentElementsDeterministically for usage
+ */
+private val divergentDocumentableComparator =
+    compareBy<Documentable, String?>(nullsLast()) { it.dri.packageName }
+        .thenBy(nullsFirst()) { it.dri.classNames } // nullsFirst for top level to be first
+        .thenBy(
+            nullsLast(
+                compareBy<Callable> { it.params.size }
+                    .thenBy { it.signature() }
+            )
+        ) { it.dri.callable }
```

Comment:
There's definitely room for improvement in terms of sorting rules, but it's a start. To me it makes sense to display top level declarations to be on top as they have a wider scope, and to sort vararg-like overloads ([like these](https://www.jooq.org/javadoc/latest/org.jooq/org/jooq/UpdateQuery.html#addValues(org.jooq.Row2,org.jooq.Row2))) in a strairs-like manner - both are implemented.

Tell me if you can think of anything else right off the bat